distribution/test/itests/test-itests-ddf/src/test/java/ddf/test/itests/catalog/TestFederation.java

```diff
@@ -1000,104 +1013,184 @@ public void testCswCreateEventEndpoint() throws Exception {
 
     /**
      * Tests basic download from the live federated csw source
-     *
      * @throws Exception
      */
     @Test
     public void testDownloadFromFederatedCswSource() throws Exception {
+        final String cometDEndpointUrl = COMETD_ENDPOINT.getUrl();
+        final String notificationsChannel = "/ddf/notifications/**";
+        final String activitiesChannel = "/ddf/activities/**";
+        final String filename = "product1.txt";
+        
+        CometDClient cometDClient = new CometDClient(cometDEndpointUrl);
+        cometDClient.start();
+        cometDClient.subscribe(notificationsChannel);
+        cometDClient.subscribe(activitiesChannel);
+        
         cswServer.whenHttp()
                 .match(Condition.get("/services/csw"),
                         Condition.parameter("request", "GetRecordById"))
-                .then(getCswRetrievalHeaders(),
-                        chunkedContentWithHeaders(STUB_SERVER_TEST_STRING,
-                                Duration.ofMillis(0),
-                                0));
+                .then(getCswRetrievalHeaders(filename), chunkedContentWithHeaders(STUB_SERVER_TEST_STRING,
+                        Duration.ofMillis(0), 0));
 
         String restUrl =
                 REST_PATH.getUrl() + "sources/" + CSW_STUB_SOURCE_ID + "/" + CSW_STUB_RESOURCE_ID
-                        + "?transform=resource";
+                        + "?transform=resource" + "&session=" + cometDClient.getClientId();
 
         // Verify that the testData from the csw stub server is returned.
         // @formatter:off
         when().get(restUrl).then().log().all().assertThat().contentType("text/plain")
                 .body(is(STUB_SERVER_TEST_STRING));
         // @formatter:on
+
+        expect("Waiting for notifications").within(10, SECONDS).until(() -> cometDClient.getMessages(notificationsChannel).size() == 1);
+        expect("Waiting for activities").within(10, SECONDS).until(() -> cometDClient.getMessages(activitiesChannel).size() == 2);
+        
+        assertThat(cometDClient.getAllMessages().size(), is(3));
+        
+        List<String> notifications = sortJsonMessagesByTimestamp(cometDClient.getMessages(notificationsChannel));
+        assertThat(notifications.size(), is(1));
+        verifyNotification(JsonPath.from(notifications.get(0)), filename, "Resource retrieval completed, 8 bytes retrieved. ", "complete");
+       
+        List<String> activities = sortJsonMessagesByTimestamp(cometDClient.getMessages(activitiesChannel));
+        assertThat(activities.size(), is(2));
+        verifyActivity(JsonPath.from(activities.get(0)), filename, "Resource retrieval started. ", "STARTED");
+        verifyActivity(JsonPath.from(activities.get(1)), filename, "Resource retrieval completed, 8 bytes retrieved. ", "COMPLETE");
+
+        cometDClient.shutdown();
```

Comment:
I see your concern. I changed the code a bit and added the shutdown call to the tearDown